sys/stats/full/src/stats.c

```diff
@@ -96,13 +92,24 @@ stats_size(const struct stats_hdr *hdr)
     return hdr->s_cnt * hdr->s_size;
 }
 
-void *
-stats_data(const struct stats_hdr *hdr)
+const void *
+stats_data_ro(const struct stats_hdr *hdr)
 {
-    size_t offset;
+    if (hdr->s_flags & STATS_HDR_F_PERSIST) {
+        return &((const struct stats_persisted_hdr *)hdr)->data[0];
+    } else {
+        return &hdr->data[0];
+    }
+}
 
-    offset = stats_offset(hdr);
-    return (uint8_t *)hdr + offset;
+void *
+stats_data_rw(struct stats_hdr *hdr)
+{
+    if (hdr->s_flags & STATS_HDR_F_PERSIST) {
+        return &((struct stats_persisted_hdr *)hdr)->data[0];
+    } else {
+        return &hdr->data[0];
+    }
```

Comment:
You might consider implementing this in terms of `stats_data_ro`.  If the stats object is not actually const (i.e., not actually located in read-only memory), then it is safe to cast away const from the result of `stats_data_ro` and return that.

I don't feel strongly about this at all.  Just throwing it out there.